_examples/google-gke-nfs-filestore/main.tf

```diff
@@ -0,0 +1,227 @@
+  ]
+
+  master_auth {
+    username = random_id.username.hex
+    password = random_id.password.hex
+  }
+
+  node_config {
+    machine_type = "n1-standard-4"
+
+    oauth_scopes = [
+      "https://www.googleapis.com/auth/compute",
+      "https://www.googleapis.com/auth/devstorage.read_only",
+      "https://www.googleapis.com/auth/logging.write",
+      "https://www.googleapis.com/auth/monitoring",
+    ]
+  }
+}
+
+
+resource "google_filestore_instance" "test" {
+  name = "test-nfs-server"
+  tier = "STANDARD"
+  zone = "us-west1-a"
+
+  file_shares {
+    capacity_gb = 1024
+    name        = "vol1"
+  }
+
+  networks {
+    network = google_compute_network.vpc.name
+    modes   = ["MODE_IPV4"]
+  }
+}
+
+data "template_file" "kubeconfig" {
+  template = file("${path.module}/kubeconfig-template.yaml")
+
+  vars = {
+    cluster_name    = google_container_cluster.primary.name
+    user_name       = google_container_cluster.primary.master_auth[0].username
+    user_password   = google_container_cluster.primary.master_auth[0].password
+    endpoint        = google_container_cluster.primary.endpoint
+    cluster_ca      = google_container_cluster.primary.master_auth[0].cluster_ca_certificate
+    client_cert     = google_container_cluster.primary.master_auth[0].client_certificate
+    client_cert_key = google_container_cluster.primary.master_auth[0].client_key
+  }
+}
+
+resource "local_file" "kubeconfig" {
+  content  = data.template_file.kubeconfig.rendered
+  filename = "${path.module}/kubeconfig"
+}
+
+provider "kubernetes" {
+  version = "1.11.2"
+  load_config_file = "false"
+
+  host = google_container_cluster.primary.endpoint
+
+  username               = google_container_cluster.primary.master_auth[0].username
+  password               = google_container_cluster.primary.master_auth[0].password
+  client_certificate     = base64decode(google_container_cluster.primary.master_auth[0].client_certificate)
+  client_key             = base64decode(google_container_cluster.primary.master_auth[0].client_key)
+  cluster_ca_certificate = base64decode(google_container_cluster.primary.master_auth[0].cluster_ca_certificate)
+}
+
+resource "kubernetes_namespace" "example" {
+  metadata {
+    name = "test"
+  }
+}
+
+resource "kubernetes_storage_class" "nfs" {
+  metadata {
+    name = "filestore"
+  }
+  reclaim_policy  = "Retain"
+  storage_provisioner = "nfs"
+}
+
+resource "kubernetes_persistent_volume" "example" {
+  metadata {
+    name = "nfs-volume"
+  }
+  spec {
+    capacity = {
+      storage = "1T"
+    }
+    storage_class_name = "filestore"
+    access_modes = ["ReadWriteMany"]
+    persistent_volume_source {
+      nfs {
+        server = google_filestore_instance.test.networks[0].ip_addresses[0]
+        path = "/${google_filestore_instance.test.file_shares[0].name}"
+      }
+    }
+  }
+}
+
+resource "kubernetes_persistent_volume_claim" "example" {
+  metadata {
+    name = "mariadb-data"
+    namespace = "test"
+  }
+  spec {
```

Comment:
It would be better to put `spec.selector` to identify what PVC is used.
For example,
```
spec {
  selector {
    match_labels = kubernetes_persistent_volume.example.metadata.labels
  }
}
```